examples/gno.land/p/demo/access/set.gno

```diff
@@ -0,0 +1,103 @@
+package access
+
+import "std"
+
+// New returns a Set object initialized with the caller as the unique authorized address.
+// It is recommended to use this function in the `init()` function of the calling realm.
+func New() Set {
+	addr := std.GetOrigCaller()
+	return Set{
+		addrs: []std.Address{addr},
+	}
+}
+
+// NewWithAddress returns a Set object initialized with the provided address as authorized.
+func NewWithAddress(addr std.Address) Set {
+	s := New()
+	s.ReplaceAll(addr)
+	return s
+}
+
+// Set is an object containing the configuration and allowing the application of filters.
+// It is suited to be used as a contract-side global variable or can be embedded in another Go object.
+type Set struct {
```

Comment:
Can we have a better naming? `Set` is a very generic name.

Here's few suggestions:

`AuthorizationSet`: Describes that it holds a list of authorized addresses.
`AuthSet`: Short for "Authorization Set", indicating its purpose of holding authorized addresses.
`AccessControl`: Suggests that it is used for controlling access to certain resources.
`WhiteList`: Suggests that it acts as a whitelist for permitted addresses.

Reply:
Or maybe? 

```diff
-type Set struct {
-	addrs []std.Address
-}
+type AuthSet []std.Address
```

Since this struct only contains one field

Reply:
What about this
```go
type AuthSet struct {
	entries []AuthSetEntry
}

type AuthSetEntry struct {
	role    Role // `Admin` etc
	address std.Address
}
```

It can provide better access control.

Reply:
The expected usage is: `access.Set` which IMO makes sense for such specialized pure packages.

However, I agree with you in other cases, especially apps or just big libraries.